homework/utils.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+/**
+ * Fire a json XMLHttpRequest.
+ *
+ * @param {string} method - HTTP methods. e.g. 'GET', 'HEAD', 'POST', 'PUT', 'DELET'... etc.
+ * @param {string} url - The endpoint of your request.
+ * @returns {Promise}
+ */
+function XHRequest(method, url) {
```

Comment:
It's good to see you refactor this part. And you have successfully "Promisified" it.